### PR TITLE
devtools: add pre-push hook check for version mismatch

### PR DESCRIPTION
**Pull request by jba**
Move existing script into a devtools directory.

Add a script that when used as a pre-push hook, will check that
the version in the code matches the latest version tag.

### CODE REVIEW

File: devtools/pre-push-hook.sh
Comment (eliben) [inline devtools/pre-push-hook.sh:1]: Will this work for merged PRs?
Maybe easier to set it up as a github action?

Comment (jba) [reply devtools/pre-push-hook.sh:1]: I want to catch this list of commands executed on a local machine. This is how I typically tag a repo:
```
git tag vX.Y.Z
git push origin vX.Y.Z
```
As far as I know, a pre-push hook is the only way to catch that.

Should I be tagging a different way? (Probably, but I still want to catch mistakes in the above.)

Comment (jba) [reply devtools/pre-push-hook.sh:1]: > Will this work for merged PRs?
I'm not sure I understand.

Comment (eliben) [reply devtools/pre-push-hook.sh:1]: My concern was that we don't typically push stuff from local, but merge a PR instead, so I was wondering how that's supposed to work. But if this is for catching tag pushes, that makes more sense!

I do tags the same way.

**Review by eliben — APPROVED** [review]

**Pull request merged**